src/data_structures/src/array_ext.cairo

```diff
@@ -135,6 +136,35 @@ impl ArrayImpl<T, impl TCopy: Copy<T>, impl TDrop: Drop<T>> of ArrayTraitExt<T>
     ) -> Option<usize> {
         self.span().index_of_max()
     }
+
+    fn dedup<impl TPartialEq: PartialEq<T>>(mut self: Array<T>) -> Array<T> {
+        if(self.len() <= 1) {
+            return self;
+        }
+
+        let mut ret = array![];
+
+        let mut last_value = self.pop_front().unwrap();
+
+        ret.append(last_value);
+
+        loop {
+            match self.pop_front() {
+                Option::Some(v) => {
+
```

Comment:
Maybe get rid of that extra line, but I'm getting picky now 😄

Reply:
Line: 154 destroyed 💣